src/app/services/loader.service.ts

```diff
@@ -0,0 +1,25 @@
+import { Injectable } from '@angular/core';
+import { Subject } from 'rxjs';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class LoaderService {
+
+  private loaderSubject = new Subject<any>();
+  loaderState = this.loaderSubject.asObservable();
+
+  constructor() {
+  }
+
+  show() {
+    this.loaderSubject.next(<any>{ show: true });
+  }
+
+  hide() {
+    setTimeout(() => {
```

Comment:
why do you need Timeout here? Is it because requests are too fast and otherwise it blinks too quick?

Reply:
yes, its annoying.  Should i delete it?

Reply:
It's up to you 😃